src/placeos-models/control_system.cr

```diff
@@ -162,31 +160,17 @@ module PlaceOS::Model
       end
     }
 
+    before_save :update_features
+
     # Adds modules to the features field,
     # Extends features with extra_features field in settings if present
     protected def update_features
-      if (id = @id)
-        system = ControlSystem.find(id)
-        if system
-          mods = system.modules || [] of String
-          mods.reject! "__Triggers__"
-          @features = mods.join " "
-        end
-      end
-
-      # TODO:
-      # Do a query for the unencrypted settings belonging to the system
-      # Append extra features
-      #
-      # if (settings = @settings)
-      #   # Extra features stored in unencrypted settings
-      #   settings.find { |(level, _)| level == Encryption::Level::None }.try do |(_, setting_string)|
-      #     # Append any extra features
-      #     if (extra_features = YAML.parse(setting_string)["extra_features"]?)
-      #       @features = "#{@features} #{extra_features}"
-      #     end
-      #   end
-      # end
+      module_names = Module
+        .find_all(@modules || [] of String)
```

Comment:
My logic is to use `@`tters in the models if you want to avoid the ORM.
So the setting of `features` below via the `@`tter, and  the use of the getter to access the `modules`